isis/src/chandrayaan1/apps/chan1m32isis/chan1m32isis.cpp

```diff
@@ -0,0 +1,480 @@
+#include "chan1m32isis.h"
+
+#include <math.h>
+
+#include <QDebug>
+#include <QString>
+
+#include "Application.h"
+#include "BoxcarCachingAlgorithm.h"
+#include "Brick.h"
+#include "Cube.h"
+#include "CubeAttribute.h"
+#include "FileName.h"
+#include "IException.h"
+#include "ImportPdsTable.h"
+#include "iTime.h"
+#include "NaifStatus.h"
+#include "OriginalLabel.h"
+#include "PixelType.h"
+#include "ProcessByLine.h"
+#include "ProcessBySample.h"
+#include "ProcessImportPds.h"
+#include "Progress.h"
+#include "Pvl.h"
+#include "Table.h"
+#include "UserInterface.h"
+
+using namespace std;
+
+namespace Isis {
+
+  void writeCube(Buffer &in);
+  void writeCubeWithDroppedLines(Buffer &in);
+  void importImage(QString outputParamName, ProcessImportPds::PdsFileType fileType);
+  void importImage(QString outputParamName, ProcessImportPds::PdsFileType fileType, UserInterface &ui);
+  void translateChandrayaan1M3Labels(Pvl &pdsLabel, Cube *ocube, Table &utcTable,
+                                     ProcessImportPds::PdsFileType fileType);
+  void flip(Buffer &in);
+  void flipUtcTable(Table &utcTable);
+  Cube *g_oCube;
+  Brick *g_oBuff;
+  int g_totalLinesAdded;
+  double g_expectedLineRate;
+  Table *g_utcTable;
+  PvlGroup g_results("Results");
+
+  void chan1m32isis(UserInterface &ui) {
+    //g_results.clear();
```

Comment:
This line results in a segfault during testing.  This needs to be fixed, but I wanted to get the PR up ASAP.  Any ideas?

Reply:
UserInterface inherits from IsisAml and the `WasEntered` method called `importImage` is inherited from IsisAml. This could be relating to the problem and the fact that in the tests, we are not using the Application class anymore.